### PR TITLE
@mzikherman - Add saveArtwork mutation

### PR DESCRIPTION
**Pull request by broskoski**
Getting a weird error with about `deprecationReason`, will figure it out shortly...

### CODE REVIEW

File: schema/me/save_artwork.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  decription: 'Save (or remove) an artwork to (from) a users default collection.',
+  inputFields: {
+    artwork_id: {
+      type: GraphQLString,
+    },
+    remove: {
+      type: GraphQLBoolean,
+    },
+  },
+  outputFields: ArtworkType.getFields(),
```
Comment (mzikherman) [inline schema/me/save_artwork.js]: ooo i didnt know you could do `getFields`. I had previously moved the fields out of the type struct (https://github.com/artsy/metaphysics/blob/20b68bc7e5f259408c123b6331b4f7eefe749568/schema/me/conversations.js#L42-L90) b/c this expects just the fields and I didn't know how to get to the fields from the type.

**Comment by mzikherman** [comment]
Yea, weird ❌ , lmk if you want to pair or when to take another look, otherwise 👍

File: schema/me/save_artwork.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
+import gravity from '../../lib/loaders/gravity';
+import { GraphQLString, GraphQLBoolean } from 'graphql';
+import { mutationWithClientMutationId } from 'graphql-relay';
```
Comment (mzikherman) [inline schema/me/save_artwork.js]: 🎉

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>Please assign someone to merge this PR, and optionally include people who should review.</td>
    </tr>
  </tbody>
</table>




<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by mzikherman** [comment]
Oh yea where da spec?? :)

**Comment by broskoski** [comment]
Coming!

File: schema/me/save_artwork.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  decription: 'Save (or remove) an artwork to (from) a users default collection.',
+  inputFields: {
+    artwork_id: {
+      type: GraphQLString,
+    },
+    remove: {
+      type: GraphQLBoolean,
+    },
+  },
+  outputFields: artworkFields(),
```
Comment (mzikherman) [inline schema/me/save_artwork.js]: oof wound up having to use the fields directly?

File: test/schema/me/save_artwork.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+  afterEach(() => {
+    SaveArtwork.__ResetDependency__('gravity');
+  });
+
+  it('saves an artwork', () => {
+    /* eslint-disable max-len */
+    const mutation = `
+      mutation {
+        saveArtwork(input: { artwork_id: "damon-zucconi-slow-verb" }) {
```
Comment (mzikherman) [inline test/schema/me/save_artwork.js]: 👍

File: schema/me/save_artwork.js (hunk 3 above)
Comment (broskoski) [inline schema/me/save_artwork.js]: Yeah, because of the whole using the Artwork.type within the artwork type definition

Comment (broskoski) [inline schema/me/save_artwork.js]: And actually because this stupid `graphql-js` library outputs fields that can't be consumed in mutation definitions.

**Pull request merged**